aladdin/commands/helm_values.py

```diff
@@ -106,3 +93,44 @@ def helm_values(
                 capture_output=False,
                 check=True,
             )
+
+
+@contextmanager
+def clone_and_checkout(repo_name, githash):
+    current_hash = None
+    current_repo = None
+    with suppress(subprocess.CalledProcessError):
+        current_repo = Git.get_repo()
+        current_hash = Git.get_hash()
+
+    if (
+        current_hash and
+        current_repo and
+        (not repo_name or current_repo == repo_name) and
+        current_hash == githash and
+        Git.clean_working_tree()
+    ):
+        yield Git.get_base_directory()
```

Comment:
this can be specially useful in CI/CD use-cases, where this will be already executed from the right repo at the right commit